src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_Tuples.cs

```diff
@@ -0,0 +1,353 @@
+                ValidateTupleNameAttribute(indexer,
+                    expectedTupleNamesAttribute: true,
+                    expectedElementNames: expectedTupleNames);
+                expectedTupleNames = new[] { "e3", "e4" };
+                ValidateTupleNameAttribute(indexer.Parameters.Single(),
+                    expectedTupleNamesAttribute: true,
+                    expectedElementNames: expectedTupleNames);
+            }
+
+            private void ValidateTupleNameAttribute(Symbol symbol,
+                bool expectedTupleNamesAttribute,
+                string[] expectedElementNames = null,
+                bool forReturnType = false)
+            {
+                var synthesizedTupleElementNamesAttr = symbol.GetSynthesizedAttributes(forReturnType)
+                    .Where(attr => string.Equals(attr.AttributeClass.Name,
+                                                 "TupleElementNamesAttribute",
+                                                 StringComparison.Ordinal))
+                    .ToList();
+                if (!expectedTupleNamesAttribute)
+                {
+                    Assert.Empty(synthesizedTupleElementNamesAttr);
+                }
+                else
+                {
+                    var tupleAttr = synthesizedTupleElementNamesAttr.Single();
+                    var expectedCtor = expectedElementNames == null
+                        ? _tupleAttrNoArgs
+                        : _tupleAttrTransformNames;
+                    Assert.NotNull(expectedCtor);
+                    Assert.Equal(expectedCtor, tupleAttr.AttributeConstructor);
+
+                    if (expectedElementNames == null)
+                    {
+                        Assert.Empty(tupleAttr.CommonConstructorArguments);
+                    }
+                    else
+                    {
+                        Assert.Equal(1, tupleAttr.CommonConstructorArguments.Length);
+
+                        var arg = tupleAttr.CommonConstructorArguments[0];
+                        Assert.Equal(TypedConstantKind.Array, arg.Kind);
+
+                        var actualElementNames = arg.Values;
+                        Assert.Equal(expectedElementNames.Length, actualElementNames.Length);
+                        var stringType = _comp.GetSpecialType(SpecialType.System_String);
+
+                        for (int i =  0; i < actualElementNames.Length; i++)
+                        {
+                            string expectedName = expectedElementNames[i];
+                            TypedConstant actualName = actualElementNames[i];
+
+                            Assert.Equal(TypedConstantKind.Primitive, actualName.Kind);
+                            Assert.Equal(stringType, actualName.Type);
+                            Assert.Equal(expectedName, (string)actualName.Value);
+                        }
+                    }
+                }
+            }
+
+        }
+    }
+}
```

Comment:
I see that (1) and (2) were addressed. Did you also add tests for (3), (4) or (5)? 
If not, please keep them in mind for a later PR. Thanks

Also, another suggestion is to check the emitted IL.